_integration-test/run.py

```diff
@@ -0,0 +1,176 @@
+import subprocess
+import os
+from functools import lru_cache
+import requests
```

Comment:
I would not recommend using requests in new code

if you can get away with it, stdlib `urllib.request` does not require additional dependencies.  httpx is a modern replacement if you find urllib.request too cumbersome

Reply:
switched to using httpx, since I wanted a good way to save cookies

Reply:
What's the reason why requests is not the preferred choice?